tests/ui/imports/issue-115858-duplicated-crate-diag.rs

```diff
@@ -0,0 +1,20 @@
+// Regression test for <https://github.com/rust-lang/rust/issues/115858>.
+// It ensures that it doesn't suggest paths like `crate::crate::unix`
+
+// edition:2018
+
+#![crate_type = "lib"]
+
+pub mod unix {
+    pub mod linux {
+        pub mod utils {
+            pub fn f() {
+                let x = crate::linux::system::Y;
+                //~^ ERROR unresolved import
```

Comment:
I didn't find a way to ensure that in the help message, we have the correct suggested path to prevent regressions from happening. If someone knows how I could do that, it'd be awesome!

Reply:
You could `// run-rustfix` which checks that the applied suggestion works

Reply:
Oh nice! Doing that right away. Thanks!

Reply:
Do you know maybe how to check this @estebank ?

Reply:
@GuillaumeGomez We should change the logic so that the last two suggestions aren't emitted if this one is. rustfix can't deal with multiple exclusive suggestions.

Reply:
You mean changing the logic behind the compiler error help messages or rustfix? In any case, what do you have in mind?

Reply:
The compiler is the one responsible to not emit unnecessary suggestions. I'd need to look at the code more closely to see how to signal that the more specific suggestion has already been added to the diagnostic, to not add the later ones.